File: backend/internal/service/check_in.go
package service

import "time"

const (
	CheckInDailyReward = 0.25
	CheckInDay4Total   = 0.45
	CheckInDay16Total  = 5.21
	CheckInMilestone4  = 4
	CheckInMilestone16 = 16
)

// UserCheckIn 用户签到记录
type UserCheckIn struct {
	ID          int64
	UserID      int64
	CheckInDate time.Time
	StreakDay   int
	DailyReward float64
	BonusReward float64
	TotalReward float64
	CreatedAt   time.Time
}

// CheckInStatus 签到状态
type CheckInStatus struct {
	CheckedInToday bool
	Streak         int
	NextMilestone  int
	TodayReward    *CheckInRewardPreview
	CalendarMonth  string
	MonthCheckIns  []UserCheckIn
}

// CheckInRewardPreview 今日可领奖励预览
type CheckInRewardPreview struct {
	DailyReward float64
	BonusReward float64
	TotalReward float64
	Milestone   string
}

// CheckInResult 签到结果
type CheckInResult struct {
	Message      string
	DailyReward  float64
	BonusReward  float64
	TotalReward  float64
	Streak       int
	NewBalance   float64
	Milestone    string
	CheckInDate  time.Time
}

File: frontend/src/api/checkIn.ts
import { apiClient } from './client'

export interface CheckInRules {
  daily_reward: number
  day4_total: number
  day16_total: number
}

export interface CheckInRecord {
  date: string
  streak_day: number
  daily_reward: number
  bonus_reward: number
  total_reward: number
}

export interface CheckInRewardPreview {
  daily_reward: number
  bonus_reward: number
  total_reward: number
  milestone?: string
}

export interface CheckInStatus {
  checked_in_today: boolean
  streak: number
  next_milestone: number
  today_reward?: CheckInRewardPreview
  rules: CheckInRules
  calendar_month: string
  month_check_ins: CheckInRecord[]
}

export interface CheckInResult {
  message: string
  daily_reward: number
  bonus_reward: number
  total_reward: number
  streak: number
  new_balance: number
  milestone?: string
  date: string
}

export async function getStatus(month?: string): Promise<CheckInStatus> {
  const { data } = await apiClient.get<CheckInStatus>('/check-in/status', {
    params: month ? { month } : undefined
  })
  return data
}

export async function checkIn(): Promise<CheckInResult> {
  const { data } = await apiClient.post<CheckInResult>('/check-in')
  return data
}

export const checkInAPI = {
  getStatus,
  checkIn
}

export default checkInAPI
